File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.cpp
#include <bits/stdc++.h>
using namespace std;

bool divisor(int num){
    long long a = sqrt(num);
    if(a * a == num)
        return false;
    return true;
}

int solution(int left, int right) {
    int answer = 0;
    for(int i = left; i <= right; i++){
        if(divisor(i)){
            answer += i;
        }
        else {
            answer -= i;
        }
    }
    return answer;
}

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.java

class Solution {
    public boolean divisor(int num){
        int check = (int)Math.sqrt(num);
        if(check * check == num)
            return false;
        return true;
    }
    
    public int solution(int left, int right) {
        int answer = 0;
        for(int i = left; i <= right; i++){
            answer += (divisor(i)) ? i : i * -1;
        }
        
        return answer;
    }
}